R/po_compile.R

```diff
@@ -0,0 +1,62 @@
+#' Compile `.po` files to `.mo`
+#'
+#' This function compiles the plain text `.po` files that translators work with
+#' into the binary `.mo` files that are installed with packages and used for
+#' live translations.
+#'
+#' @param dir Path to package root directory.
+#' @param package Name of package. If not supplied, read from `DESCRIPTION`.
+#' @param lazy If `TRUE`, only `.mo` functions that are older than `.po`
+#'   files be updated
+#' @param verbose If `TRUE`, print information as it goes.
+po_compile = function(dir = ".", package = NULL, lazy = TRUE, verbose = TRUE) {
+  po_metadata <- get_po_metadata(dir = dir, package = package)
+  dir_create(dirname(po_metadata$mo))
+
+  if (lazy) {
+    po_metadata <- po_metadata[is_outdated(po_metadata$po, po_metadata$mo)]
+  } else {
+    # Clear out all older translations
+    mo_dirs <- dir(file.path(dir, "inst", "po"), full.names = TRUE)
+    to_delete <- mo_dirs[!basename(mo_dirs) %in% c(po_metadata$language, "en@quot")]
+
+    for (dir in to_delete) {
+      if (verbose) messagef(
+        "Found a compiled translation for %s at %s, but no corresponding .po; deleting",
+        basename(dir), dirname(dir)
+      )
+      unlink(dir, recursive = TRUE)
+    }
+  }
+
+  for (ii in seq_len(nrow(po_metadata))) {
+    row <- po_metadata[ii]
+    if (verbose) messagef("Recompiling '%s' %s translation", row$language, row$type)
```

Comment:
ditto here: one message instead of many

Reply:
I think you definitely want multiple messages here since `msgfmt` produces information about each file which is otherwise contextless. 

I think that'll be more obvious once we have some snapshot tests.

Reply:
Oh, right, in this case it won't look like

```
Recompiling 'es' R translation
Recompiling 'es' src translation
Recompiling 'ja' R translation
...
```

but instead

```
Recompiling 'es' R translation
# <msgfmt output>
Recompiling 'es' src translation
# <msgfmt output>
Recompiling 'ja' R translation
# <msgfmt output>
...
```

Agree it's good as is